### PR TITLE
Bringing the OOMMFC package to conda-forge

### PR DESCRIPTION
**Pull request by fangohr**
The OOMMF Calculator (OOMMFC) is a calculator for the numerical
operations required for micromagnetic calculations. Technically,
the OOMMFC class inherits from the "micromagneticmodel", which is
an abstract description of a micromagnetic problem.
    
Another way of describing OOMMFC is that it provides a Python
interface to the OOMMF micromagnetic package from NIST.

This is the second last package from the Jupyter-OOMMF project
(http://joommf.github.io)

### CODE REVIEW

**Comment by fangohr** [comment]
@takluyver , @mb1a15 , @rpep - the windows build for `oommfc` (this pull request) fails (see for example https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.14305/job/a7a1m9inioi8rhj6). It looks to me that the code is trying to run OOMMF through docker rather than the native install (which we provide with the oommf condo-forge package). Could you have a look at this and advise, please?

**Comment by takluyver** [comment]
The checks it does to use our conda oommf package on Windows are:

- `sys.platform == 'win32'` : check for Windows
- `('Continuum' in sys.version) or ('Anaconda' in sys.version)` : check for conda
- `{sys.prefix}\opt\oommf` is a directory : check that our oommf package is installed.

At least one of those checks must be failing. Can you insert some test steps before we call `oommfc.test()` in order to work out which one?

**Comment by isuruf** [comment]
Second check will fail as the version string has "Packaged by conda-forge", but no "Continuum" or "Anaconda"

**Comment by takluyver** [comment]
Ah, thanks. Do you know of a better way to check whether the Python running is in a conda environment? Or should we just add `conda-forge` to the strings we check in `sys.version`?

**Comment by isuruf** [comment]
For newer versions of conda, you can check whether `sys.prefix` directory has a file named `conda.exe or conda.bat or conda`

**Comment by takluyver** [comment]
Do I recall rightly that conda only used to install itself in the root env, not other envs you create? Do you know which version started putting the executable in every environment?

**Comment by takluyver** [comment]
I notice that all my conda envs appear to have a `conda-meta` directory directly under the prefix - is that reliable?

**Comment by isuruf** [comment]
I think so, but I'm not sure

**Comment by takluyver** [comment]
OK, I've made it check for `conda-meta` instead, but we'll need to do a new release and update the recipe.

**Comment by fangohr** [comment]
Dear @isuruf, thank you for your help (above) - it took us some time, but now the recipe builds, and is the last step in getting the JOOMMF software onto conda. Could you have a look at this again, please? (@takluyver, @marijanbeg).

File: recipes/oommfc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - python
+    - pip
+  run:
+    - python
+    - sarge
+    - scipy
+    - micromagneticmodel
+    - oommfodt
+    - oommf
+    - pytest
```
Comment (isuruf) [inline recipes/oommfc/meta.yaml]: `pytest` should be in `test: requires:` as it is not a run dependency

Comment (fangohr) [reply recipes/oommfc/meta.yaml]: Thanks, but in this case it is actually a run time requirement: there is a function `oommfc.test()` that can be run after importing the module which will use `purest` to run the test suite of the package.

Comment (takluyver) [reply recipes/oommfc/meta.yaml]: I think we had a similar discussion on one of our other packages before. It's debatable whether that counts as a runtime requirement or not, but we've left it as such in the discretisedfield package, for example:

https://github.com/conda-forge/discretisedfield-feedstock/blob/0733df2ee365de503c678523a900497180a4f077/recipe/meta.yaml#L30

Comment (isuruf) [reply recipes/oommfc/meta.yaml]: Okay

**Comment by fangohr** [comment]
Thank you!

**Pull request merged**